### PR TITLE
fix(profile-sync-controller): handle non-JSON error responses in authentication services

### PR DESCRIPTION
**Pull request by mathieuartu**
## Explanation

Related to: https://github.com/MetaMask/metamask-extension/issues/37080

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [x] I've communicated my changes to consumers by [updating changelogs for packages I've changed](https://github.com/MetaMask/core/tree/main/docs/processes/updating-changelogs.md)
- [ ] I've introduced [breaking changes](https://github.com/MetaMask/core/tree/main/docs/processes/breaking-changes.md) in this PR and have prepared draft pull requests for clients and consumer packages to resolve them

---

> [!NOTE]
> Unifies and hardens error handling in authentication services to prevent crashes on non-JSON responses and provide consistent, informative errors.
> 
> - Introduces `throwServiceError` with `getResponseErrorMessage` and `isErrorResponse`; parses `Retry-After` and throws `RateLimitedError` on 429
> - Refactors `getNonce`, `authenticate`, `authorizeOIDC`, `pairIdentifiers`, and `getUserProfileLineage` to use the new helper and standardized error prefixes/messages
> - Adds comprehensive tests for JSON/non-JSON errors, edge cases (null/undefined thrown values), and `Retry-After` variations
> - Minor typings/return type annotations for URL builders; updates `CHANGELOG.md` accordingly
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 8195c5d6dcf27d68832a6e2a6a2edc3af26619f6. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: packages/profile-sync-controller/src/sdk/authentication-jwt-bearer/services.ts [hunk 1]
```diff
@@ -216,9 +280,9 @@ export async function authorizeOIDC(
   platform: Platform,
 ): Promise<AccessToken> {
   const grantType = 'urn:ietf:params:oauth:grant-type:jwt-bearer';
-  const headers = new Headers({
```
Comment (gantunesr) [inline packages/profile-sync-controller/src/sdk/authentication-jwt-bearer/services.ts:219]: Why not keep it as `new Headers`?

File: packages/profile-sync-controller/src/sdk/authentication-jwt-bearer/services.ts [hunk 2]
```diff
@@ -242,15 +310,12 @@ export async function authorizeOIDC(
...
-      throw e;
-    }
-    /* istanbul ignore next */
-    const errorMessage =
-      e instanceof Error ? e.message : JSON.stringify(e ?? '');
-    throw new SignInError(`unable to get access token: ${errorMessage}`);
+  } catch (error) {
+    return await throwServiceError(
+      error,
+      'Unable to get access token',
```
Comment (gantunesr) [inline packages/profile-sync-controller/src/sdk/authentication-jwt-bearer/services.ts]: Do you think we can have all messages following the same format? Something like `Unable to X`, `Failed to X`, or `Error: Could not X`

File: packages/profile-sync-controller/src/sdk/authentication-jwt-bearer/services.ts (hunk 1 above)
Comment (mathieuartu) [reply packages/profile-sync-controller/src/sdk/authentication-jwt-bearer/services.ts:219]: It fell under an eslint rule about using `fetch` types that do not exist in the NodeJS realm. It was already existing, but no updates were made to the package since, so I had to update it within this PR's scope

File: packages/profile-sync-controller/src/sdk/authentication-jwt-bearer/services.ts (hunk 2 above)
Comment (mathieuartu) [reply packages/profile-sync-controller/src/sdk/authentication-jwt-bearer/services.ts]: Nice suggestion! Added in [a1803e4](https://github.com/MetaMask/core/pull/7721/commits/a1803e40009d23b227571375b8c06fb635a6e232)
